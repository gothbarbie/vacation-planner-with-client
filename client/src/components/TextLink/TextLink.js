// @flow

import React from 'react'
import type { Element } from 'react'

import './TextLink.css'

type props = {
  children: Element<any>,
  url: string,
}

const TextLink = ({ children, url }: props) => (
  <a className="text-link" href={url}>
    {children}
  </a>
)

export default TextLink
