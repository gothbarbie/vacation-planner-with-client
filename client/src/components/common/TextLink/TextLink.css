.text-link:link, .text-link:visited {
  text-decoration: none;
}
.text-link:active, .text-link:hover {
  text-decoration: underline;
}
