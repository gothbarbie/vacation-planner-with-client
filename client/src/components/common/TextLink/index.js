// @flow

import React from 'react'
import type {Â Element } from 'react'


import './TextLink.css'

type props = {
  children: Element<any>,
  url: string,
}

export default ({ children, url }: props) => (
  <a className="text-link" href={url}>
    {children}
  </a>
)
