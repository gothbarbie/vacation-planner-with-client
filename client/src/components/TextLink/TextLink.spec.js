import React from 'react'
import { shallow } from 'enzyme'

import TextLink from './TextLink'

describe('components/TextLink', () => {
  const textLink = shallow(<TextLink />)

  it('renders', () => {
    expect(textLink).toMatchSnapshot()
  })
})
