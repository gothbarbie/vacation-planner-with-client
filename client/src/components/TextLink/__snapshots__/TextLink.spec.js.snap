// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TextLink renders 1`] = `
<a
  className="text-link"
/>
`;
